modules/new_profile/php/profile.class.inc

```diff
@@ -120,8 +135,128 @@ class Profile extends \NDB_Page
                 ->withStatus(405)
                 ->withHeader("Allow", "GET,POST");
         }
+    }
 
+    /**
+     * Verifies the values from POST message.
+     *
+     * @param array $values the POST data received from frontend form.
+     *
+     * @return array the verified parsed POST response.
+     */
+    function processValidity(array $values): array
+    {
+        // Validate (dateTaken, dateTakenConfirm, sex) not set.
+        if (!isset($values['dateTaken'])
+            || !isset($values['dateTakenConfirm'])
+            || !isset($values['sex'])
+        ) {
+            return array('error' => 'Incomplete candidate fields.');
```

Comment:
this is only checking for if dob and sex fields are set but all the fields are required on the front-end so shouldn't they also all be required in the back-end?

Reply:
except for edc because that depends on configuration

Reply:
And it would be good to add an EDC check here too if LORIS is configured that way.

Reply:
true!